### DISCUSSION TITLE
Faulty openssl crypto library error while running on macOS

### DISCUSSION THREAD

**Discussion by prachi220** [stefanberger/swtpm#851] [Q&A]
Hi,
I'm trying to run the simulator on an M1 Mac using the instructions in the Wiki for OSx. The paths are not very clear but this is how I'm trying to run the configure cmd:
`LDFLAGS="-L/usr/local/opt/openssl/lib -L/usr/local/opt/openssl/include" CFLAGS="-I/usr/local/opt/openssl/include -I/usr/local/opt/openssl/include -O2" CC=gcc PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig ./autogen.sh --prefix=/Users/<myusername> --with-openssl --with-tss-user=<myusername> --with-tss-group=<myusername>`

This ends up with an `configure: error: Faulty openssl crypto library` despite having openssl correctly installed in my system. 

```$ brew info openssl
==> openssl@3: stable 3.3.0 (bottled)
Cryptography and SSL/TLS Toolkit
https://openssl.org/
Installed
/usr/local/Cellar/openssl@3/3.3.0 (6,976 files, 32.8MB) *
  Poured from bottle using the formulae.brew.sh API on 2024-05-04 at 21:24:24
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/o/openssl@3.rb
License: Apache-2.0
==> Dependencies
Required: ca-certificates ✔
==> Caveats
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /usr/local/etc/openssl@3/certs

and run
  /usr/local/opt/openssl@3/bin/c_rehash
==> Analytics
install: 385,015 (30 days), 1,070,301 (90 days), 4,468,307 (365 days)
install-on-request: 54,867 (30 days), 137,304 (90 days), 657,883 (365 days)
build-error: 3,190 (30 days)
```

I need some help trying to figure out what could be the reason behind this issue?

**Comment by stefanberger** [comment]
The error appears due to this check here:
```
        AC_CHECK_LIB(crypto,
                      [AES_set_encrypt_key],
                      [true],
                      AC_MSG_ERROR(Faulty openssl crypto library))
```
It's checking for a symbol in the crypto library but cannot find it ...

Can you run this type of command with your OpenSSL 3.3 library:
```
# strings /usr/lib64/libcrypto.so | grep AES_set
AES_set_encrypt_key
AES_set_decrypt_key
```

I still see these symbols but maybe they disappeared.

I just looked on a Mac with OpenSSL 3.3.0 installed and there I can run the following (and also build libtpms and swtpm):

```
# strings /usr/local/lib/libcrypto.a | grep AES_set
_AES_set_encrypt_key
_AES_set_decrypt_key
```

I don't find useful symbols in libcrypto.dylib so this should be the file to look at and it seems to have the symbols that we are looking for.
